addons/im_livechat/models/im_livechat_channel_member_history.py

```diff
@@ -0,0 +1,18 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, fields
+
+
+class ImLivechatChannelMemberHistory(models.Model):
+    _name = "im_livechat.channel.member.history"
+    _description = "Keep the channel member history"
+
+    channel_id = fields.Many2one("discuss.channel", ondelete="set null")
```

Comment:
Channel should be required and cascade, it makes no sense to keep history of a removed channel.

Reply:
I think it still makes sense to keep it. The point of this model is to keep track of channel history for statistics. Even if the channel is deleted, the record can still be useful. E.g. number of conversations for a given agent.

Reply:
I don't think we want to count a deleted conversation as a conversation for somebody.

Other fields make sense to keep the history but not for channel.